From 6a35574b237321ed813d682a7702ef813c8810ac Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Wed, 27 Dec 2023 04:49:20 +0200
Subject: [PATCH] [SW-0] gaudi3: security: use correct api

This patch uses the proper ppboot api to check whether
real or fake security is enabled or not and decide whether
to open pcie registers using range registers. Earlier efuse
driver api was used which was checking only real security
is enabled or not and opening the pcie registers using range
registers even in case if fake security was enabled.

Change-Id: I2bc9fa365f8c5f71b66cf531fdc5380ef02d98fe
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 06ff4fd..90150c6 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1301,15 +1301,8 @@
 
 void hl_dev_security_init(void)
 {
-	u32 val;
-	int err;
-
-	err = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &val);
-	if (err)
-		return;
-
-	/* TODO: Do meaningful security config incase SEC_EN is set */
-	if (val & EFUSE_SEC_EN_MASK)
+	/* TODO: implement efuse driver functionality */
+	if (ppboot_sys_cfg_efuse_is_set(EFUSE_SEC_EN_MASK))
 		return;
 
 	security_rr_open_pcie_config();
